src/sentry/plugins/providers/integration_repository.py

```diff
@@ -33,37 +34,46 @@ def dispatch(self, request, organization, **kwargs):
         except Exception as e:
             return self.handle_api_error(e)
 
-        try:
-            with transaction.atomic():
-                repo = Repository.objects.create(
-                    organization_id=organization.id,
-                    name=result["name"],
-                    external_id=result.get("external_id"),
-                    url=result.get("url"),
-                    config=result.get("config") or {},
-                    provider=self.id,
-                    integration_id=result.get("integration_id"),
-                )
-        except IntegrityError:
-            # Try to delete webhook we just created
+        repo_update_params = {
+            "external_id": result.get("external_id"),
+            "url": result.get("url"),
+            "config": result.get("config") or {},
+            "provider": self.id,
+            "integration_id": result.get("integration_id"),
+        }
+
+        # first check if there is a repository without an integration that matches
+        repo = Repository.objects.filter(
+            organization_id=organization.id, name=result["name"], integration_id=None
```

Comment:
will this ever be an 'old' github repo? like `provider = github` instead of `provider = integrations:github` ? And if it is, is that fine, or would we want this to filter on the provider being `None`?

Reply:
@MeredithAnya  Great question. Actually, it could be but I think pulling it into the integration from the plugin is also fine. Probably better to enable this case IMO even though upgrading should also handle it.

Reply:
@scefali okays, maybe we can just add some logging in there so we have some way of knowing that we updated the repo (whether it was a legacy provider or none at all). Thoughts?

Reply:
@MeredithAnya yea, adding logging sounds good